service/interpreter/stateExecutionCounter.go

```diff
@@ -133,10 +188,18 @@ func (e *StateExecutionCounter) updateStateIdSearchAttribute() error {
 // ClearExecutingStateIdsSearchAttributeFinally should only be called at the end of workflow
 func (e *StateExecutionCounter) ClearExecutingStateIdsSearchAttributeFinally() {
 	config := e.configer.Get()
-	if config.GetDisableSystemSearchAttribute() {
-		return
+
+	if e.globalVersioner.IsAfterVersionOfExecutingStateIdMode() {
+		if config.GetExecutingStateIdMode() == "DISABLED" {
+			return
+		}
+	} else {
+		if config.GetDisableSystemSearchAttribute() {
+			return
+		}
 	}
```

Comment:
I may miss something, do we need this block? If the version is not after, the next block will not do anything ( `!e.globalVersioner.IsAfterVersionOfExecutingStateIdMode()` )

It seems like we just need to keep
```
if config.GetDisableSystemSearchAttribute() {
		return
```

Reply:
btw, you can consider this as a Nit comment. It is not wrong, but just not necessary :D